src/main/java/com/redhat/sast/api/service/OshSchedulerService.java

```diff
@@ -0,0 +1,573 @@
+package com.redhat.sast.api.service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.redhat.sast.api.config.OshConfiguration;
+import com.redhat.sast.api.config.OshRetryConfiguration;
+import com.redhat.sast.api.enums.OshFailureReason;
+import com.redhat.sast.api.model.Job;
+import com.redhat.sast.api.model.OshSchedulerCursor;
+import com.redhat.sast.api.model.OshUncollectedScan;
+import com.redhat.sast.api.repository.OshSchedulerCursorRepository;
+import com.redhat.sast.api.v1.dto.osh.OshScanResponse;
+
+import io.quarkus.scheduler.Scheduled;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.transaction.Transactional;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Scheduler service for automated OSH (Open Scan Hub) scan discovery and job creation.
+ *
+ * This service implements a two-phase automation loop:
+ *
+ * PHASE 1 - Incremental Scan Processing:
+ * 1. Polls OSH for new completed scans at regular intervals
+ * 2. Filters scans by configured package list
+ * 3. Creates SAST-AI workflow jobs for eligible scans
+ * 4. Maintains cursor state for incremental polling
+ * 5. Records failures for retry processing
+ *
+ * PHASE 2 - Retry Processing (if retry enabled):
+ * 1. Fetches failed scans from retry queue
+ * 2. Attempts retry processing with backoff timing
+ * 3. Removes successful retries from queue
+ * 4. Updates retry attempt counters for failures
+ *
+ * Both phases handle failures gracefully without stopping the scheduler.
+ */
+@ApplicationScoped
+@Slf4j
+public class OshSchedulerService {
```

Comment:
I don't think this is a Service class. It's a scheduler. Move this into `startup` package. That make more sense.